geode-cq/src/distributedTest/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceConstraintsDistributedTest.java

```diff
@@ -72,158 +90,74 @@
   private static final int REPLACE_KEY = ENTRIES - 4;
   private static final int INVALIDATE_KEY = ENTRIES - 5;
   private static final QueryObject TEST_VALUE = new QueryObject(999, "name_999");
-  private File logFile;
-  protected MemberVM server;
-  protected ClientVM client;
-  private static TestCqListener cqListener = null;
+
+  private static volatile CountingCqListener cqListener;
+  private static volatile ServerLauncher serverLauncher;
+  private static volatile ClientCache clientCache;
+
+  private VM serverVM;
+  private VM clientVM;
+
+  private String regionName;
+  private String queryString;
 
   @Rule
-  public ClusterStartupRule cluster = new ClusterStartupRule();
+  public DistributedRule distributedRule = new DistributedRule();
```

Comment:
I thought we were encouraging people to use the `ClusterStartuoRule` instead of `DistributedRule`... did that change?.

Reply:
Actually, we're encouraging people t use DistributedRule with Geode APIs instead of ClusterStartupRule. ClusterStartupRule provides an entire new API layer for Geode and I'm 100% against that approach for writing tests.

Reply:
Okay, good to know this... so, as the rule of thumb, we should use `DisributedRule` and `CacheRule` for distributed tests, and `ServerStarterRule` / `LocatorStarterRule` for integration tests, is that right?.

Reply:
I would just `DistributedRule` for distributed tests. Don't even use `CacheRule`. I set ServerLauncher and LocatorLauncher statics or fields in the test for starting Server and Locator. These are the same APIs that a User would use (or GFSH).